### ISSUE TITLE
Javascript errors inside markdown should return status code 1 (currently the error is ignored)

### ISSUE THREAD

**Issue by edi9999** [prettier/prettier#11908]
**Environments:**

- Prettier Version: 2.5.0
- Running Prettier via: CLI
- Runtime: node js 16.13.1
- Operating System: Linux
- Prettier plugins (if any): none

**Steps to reproduce:**

With JS files, an error will throw and result in exit code 1 :

```
$ ./node_modules/.bin/prettier foo.js
foo.js[error] foo.js: SyntaxError: Unexpected token (2:1)
[error]   1 | a = {
[error] > 2 |
[error]     | ^
```

With Markdown files, the same error will be ignored when parsing JS code that is inside a triple backtick :

````
$ ./node_modules/.bin/prettier foo.md
Hello !

```js
a = {
```
````

**Expected behavior:**

I would expect the second command with foo.md to also fail

**Actual behavior:**

The command with the md file succeeds.

**Comment by fisker** [comment]
This behavior is expected.

**Comment by alexander-akait** [comment]
@fisker Make sense to have it, maybe we need to add the new option for CLI, like --embedded-language-exit-code-on-invalid`, but very long name...

**Comment by edi9999** [comment]
Maybe something like : 

--embedded-language-error-exit
--embedded-language-bail-on-error
--embedded-language-passthrough-errors

**Comment by edi9999** [comment]
My internal usecase is that I'm writing documentation for a JS library (docxtemplater), and I use prettier on the documentation part, which also has some code samples.

I found out that some code blocks are not indented (because of a parse error), but I had no idea that there was such an error.

**Comment by alexander-akait** [comment]
Yep, I am understanding

**Comment by fisker** [comment]
Why not use `eslint-plugin-markdown`?

**Comment by fisker** [comment]
> Make sense to have it.

I don't think so, some example in markdown mean to be broken.

**Comment by alexander-akait** [comment]
Let's wait answer, maybe you are right, and better to use linter for this

**Comment by fisker** [comment]
BTW, it's already possible to throw that error, `PRETTIER_DEBUG=1 prettier *` will throw.

**Comment by edi9999** [comment]
Yes, I could probably be using eslint-plugin-markdown for that, I didn't thought about it.

**Comment by thorn0** [comment]
Previous discussion: #8488
